core/pbcc_subscribe_event_listener.c

```diff
@@ -589,13 +650,27 @@ void pbcc_object_listener_free_(pbcc_object_listener_t* listener)
 pbcc_listener_t* pbcc_listener_alloc_(
     const void*                               subscription,
     const pubnub_subscribe_listener_type      type,
-    const pubnub_subscribe_message_callback_t callback)
+    const pubnub_subscribe_message_callback_t callback,
+    void*                                     user_data)
 {
     PBCC_ALLOCATE_TYPE(listener, pbcc_listener_t, true, NULL);
     listener->counter             = pbref_counter_alloc();
     listener->type                = type;
     listener->subscription_object = subscription;
     listener->callback            = callback;
+    listener->user_data           = user_data;
+
+    return listener;
+}
+
+pbcc_status_listener_t* pbcc_status_listener_alloc_(
+    pubnub_subscribe_status_callback_t callback,
+    void*                              user_data)
+{
+    PBCC_ALLOCATE_TYPE(listener, pbcc_status_listener_t, true, NULL);
+    listener->counter   = pbref_counter_alloc();
```

Comment:
I see counter, but was it used anywhere? For `pbcc_listener_t` it is used with `pbcc_add_listener_` and `pbcc_remove_listener_`.